tests/mir-opt/funky_arms.rs

```diff
@@ -1,6 +1,5 @@
 // skip-filecheck
 // EMIT_MIR_FOR_EACH_PANIC_STRATEGY
-//@ compile-flags: --crate-type lib -Cdebug-assertions=no
```

Comment:
Why did you remove `--crate-type lib`?

Reply:
mir-opt tests are `--crate-type=rlib` by default since https://github.com/rust-lang/rust/pull/120060. codegen tests should probably get the same treatment; it's a much better default.